### PR TITLE
[NEW] GaussDB for MySQL

### LINKED ISSUE #451: [NEW] GaussDB for MySQL
https://confluence.tsi-dev.otc-service.com/display/DR/RM-4035-GAUSSDB-API?sortBy=date&sortOrder=descending

part of https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/issues/1805

### PR DESCRIPTION
**Pull request by Aloento**
fix https://github.com/opentelekomcloud/gophertelekomcloud/issues/451

https://docs.otc.t-systems.com/gaussdb-mysql/api-ref/api/index.html

### CODE REVIEW

File: openstack/gaussdb/v3/backup/CreateGaussMySqlBackup.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	var res CreateGaussMySqlBackupResponse
+	err = extract.Into(raw.Body, &res)
+	return &res, err
+}
+
+type CreateGaussMySqlBackupResponse struct {
+	Backup Backup `json:"backup"`
+	// 任务ID。
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/backup/CreateGaussMySqlBackup.go]: pls translate

File: openstack/gaussdb/v3/backup/CreateGaussMySqlBackup.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	// Backup name
+	Name string `json:"name"`
+	// Backup description
+	Description string `json:"description"`
+	// Backup start time in the "yyyy-mm-ddThh:mm:ssZ" format,
+	// where "T" indicates the start time of the time field, and "Z" indicates the time zone offset.
+	BeginTime string `json:"begin_time"`
+	// Backup status
+	// Valid value:
+	// 	BUILDING: Backup in progress
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/backup/CreateGaussMySqlBackup.go]: unicode symbols, pls remove

File: openstack/gaussdb/v3/instance/ChangeGaussMySqlInstanceSpecification.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+type MysqlChangeSpecificationOpts struct {
+	// Instance ID, which is compliant with the UUID format.
+	InstanceId string
+	// Specification change information.
+	ResizeFlavor MysqlResizeFlavor `json:"resize_flavor"`
+	// Whether the order will be automatically paid after yearly/monthly instances are changed.
+	// 	true: The order will be automatically paid from your account. The default value is true.
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/instance/ChangeGaussMySqlInstanceSpecification.go]: unicode symbols

File: openstack/gaussdb/v3/instance/DeleteGaussMySqlInstance.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+// DeleteGaussMySqlInstance
+// return ID of DB instance deletion task
+func DeleteGaussMySqlInstance(client *golangsdk.ServiceClient, instanceId string) (string, error) {
+	// DELETE https://{Endpoint}/mysql/v3/{project_id}/instances/{instance_id}
+	raw, err := client.Delete(client.ServiceURL("instances", instanceId), &golangsdk.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return extraJob(err, raw)
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/instance/DeleteGaussMySqlInstance.go]: again maybe better not return task id, but wait for completion

File: openstack/gaussdb/v3/instance/DeleteGaussMySqlReadonlyNode.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+)
+
+func DeleteGaussMySqlReadonlyNode(client *golangsdk.ServiceClient, instanceId string, nodeId string) (string, error) {
+	// DELETE https://{Endpoint}/mysql/v3/{project_id}/instances/{instance_id}/nodes/{node_id}
+	raw, err := client.Delete(client.ServiceURL("instances", instanceId, "nodes", nodeId), &golangsdk.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return extraJob(err, raw)
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/instance/DeleteGaussMySqlReadonlyNode.go]: and here

File: openstack/gaussdb/v3/instance/UpdateGaussMySqlInstanceName.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+func UpdateGaussMySqlInstanceName(client *golangsdk.ServiceClient, opts MysqlUpdateInstanceNameOpts) (string, error) {
+	b, err := build.RequestBody(opts, "")
+	if err != nil {
+		return "", err
+	}
+
+	// PUT https://{Endpoint}/mysql/v3/{project_id}/instances/{instance_id}/name
+	raw, err := client.Put(client.ServiceURL("instances", opts.InstanceId, "name"), b, nil, nil)
+	return extraJob(err, raw)
```
Comment (anton-sidelnikov) [inline openstack/gaussdb/v3/instance/UpdateGaussMySqlInstanceName.go]: and here

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
and also tests at least unit, if api not in working state

**Review by artem-lifshits — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**